js/flightlog_fields_presenter.js

```diff
@@ -778,19 +962,47 @@ function FlightLogFieldPresenter() {
                     case 'debug[2]':
                         return value.toFixed(0) + "\u03BCs";
                     }
-                    break;
                 case 'FFT_FREQ':
                     switch (fieldName) {
-                    case 'debug[2]': // pre-dyn notch gyro [for selected axis]
-                    case 'debug[3]': // raw gyro [for selected axis]
+                    case 'debug[3]': // raw gyro [for debug axis]
                         return Math.round(flightLog.gyroRawToDegreesPerSecond(value)) + "deg/s";
                     default:
                         return value.toFixed(0) + "Hz";
                     }
+                case 'RTH':
+                    switch(fieldName) {
+                        case 'debug[1]':
+                            return (value / 100).toFixed(1) + 'deg';
+                        default:
+                            return value.toFixed(0);
+                    }
+                case 'ITERM_RELAX':
+                    switch (fieldName) {
+                        case 'debug[0]': // roll setpoint high-pass filtered
+                            return value.toFixed(0) + 'deg/s';
+                        case 'debug[1]': // roll I-term relax factor
+                            return value.toFixed(0) + '%';
+                        case 'debug[3]': // roll absolute control axis error
+                            return (value / 10).toFixed(1) + 'deg';
+                    }
+                case 'RC_SMOOTHING':
+                    switch (fieldName) {
+                        case 'debug[0]':
+                            return (value + 1500).toFixed(0) + " us";
+                        case 'debug[3]': // rx frame rate [us]
+                            return (value / 1000).toFixed(1) + 'ms';
+                    }
```

Comment:
Sonarcloud is complaining about this one and other ones. This has loose ends. What happens if it's one of the other values from https://github.com/betaflight/blackbox-log-viewer/pull/549/files#diff-7c4c7dfe7836c00cb677b231aa1eba3d39c8ab2c13a82f5e62bc840a9e12d469R419-R425 ?

E.G. if `fieldName` is `debug[1]` or `debug[all]` this will fall through/have nothing.

Reply:
I understood that the intent was to let any un-specified lines fall through, and then be caught at the end, at 1084:
```
            }
             return value.toFixed(0);
         }
```

Reply:
You can use the `default:` case
```
                    switch (fieldName) {
                        case 'debug[0]':
                            return (value + 1500).toFixed(0) + " us";
                        case 'debug[3]': // rx frame rate [us]
                            return (value / 1000).toFixed(1) + 'ms';
                        default:
                            return value.toFixed(0);
                    }
```

Reply:
No, not that I can see.
You guys are absolutely right.
I just now did some basic testing with an online javascript code tester; we fall through only for the main `debugMode` case.  A failure of a debug match must be caught.  
I will use the default method you show here, it works.

Reply:
I've addressed all the code smells from trailing spaces, and put a default fallback in every switch condition, even those where all four are defined in case something else stuffs them up.  There were quite a lot of historical errors of this kind, thanks for encouraging me to fix them.

I think I've fixed all the code 'smells' that need fixing.

I can't figure how to fix the code smell `Unexpected string concatenation` in the long calculations that return both RPM and Hz.  Not sure if it matters, works fine as is, and is a 'one-off' bit of code.  I don't know how to make a function to return that stuff but if it mattered, perhaps someone else could make a PR to do that after this is merged.

Even though it complains about some of the switch statements only having one case, I've left them to all use the same method throughout since consistency seems more important, and it simplified later adjustments.

Likewise the complaints about too many lines in a function or file I can't really fix and don't think we should care too much.